src/plugins/workspace/public/plugin.ts

```diff
@@ -74,11 +77,36 @@ export class WorkspacePlugin implements Plugin<{}, {}> {
     });
   }
 
-  public async setup(core: CoreSetup, { savedObjectsManagement }: WorkspacePluginSetupDeps) {
+  /**
+   * If workspace is enabled and user has entered workspace, hide advance settings and dataSource menu and disable
+   */
+  private disableManagementApps(core: CoreSetup, management: ManagementSetup) {
+    const currentWorkspaceId$ = core.workspaces.currentWorkspaceId$;
+    this.managementCurrentWorkspaceIdSubscription?.unsubscribe();
+
+    this.managementCurrentWorkspaceIdSubscription = currentWorkspaceId$.subscribe(
+      (currentWorkspaceId) => {
+        if (currentWorkspaceId) {
+          const managementSectionApps = management.sections.section.opensearchDashboards.getAppsEnabled();
+          const disabledApps = managementSectionApps.filter(
+            (app) => app.id === 'settings' || app.id === 'dataSources'
+          );
+          disabledApps?.forEach((app) => app.disable());
+        }
+      }
+    );
+  }
```

Comment:
I think we can simplify it like this:
```suggestion
  private disableManagementApps(management: ManagementSetup) {
    ['settings', 'dataSources'].forEach((appId) =>
      management.sections.section.opensearchDashboards.getApp(appId)?.disable()
    );
  }
```

Then put `this.disableManagementApps(management)` in `if (workspaceId) {}` on line 118

Reply:
I used to do this way.  But @SuZhou-Joe commented to let me do this change. https://github.com/ruanyl/OpenSearch-Dashboards/pull/325#discussion_r1560708338. Would be better if you can align.

Reply:
@SuZhou-Joe Do you have concern on this way? I think that's fine as page always refreshed when entering new workspace

Reply:
@raintygao What's your opinion about this?

Reply:
I'm okay to both ways. Subscribing approach seem more safety, but my previous approach may also have no loopholes and looks more straightforward.

Reply:
Actually I just feel like we should has only one single source of truth. And for plugins and other logic within workspace, they should rely on currentWorkspaceId$ to detect if inside a workspace instead of using a intermediate state.

Reply:
> And for plugins and other logic within workspace, they should only rely on currentWorkspaceId$ to detect if inside a workspace instead of using a intermediate state.

@SuZhou-Joe Yes, I agree with this. But does it necessary to enforce this within workspace plugin? Considering workspace plugin is actually the place "set" this single source of truth. btw, I don't have a strong opinion on this.

@raintygao I've updated my suggested changes, please take a look if that makes sense.

Reply:
Thank you guys. I think both approaches are good and have their own reasons.

@ruanyl Yes, your suggested changes are more concise and straightforward. Thank you.